### DISCUSSION TITLE
Удалённый доступ.

### DISCUSSION THREAD

**Discussion by Crezidarkman** [vvip-68/LedPanelWiFi#19] [Вопросы и ответы]
Задачка... Есть рабочая гирлянда на окне версия 14. Имею статический ip, делаю проброс по 80 порту. Пытаюсь зайти. Прогружается стартовая страничка, а в центре постоянно сообщение "идет загрузка". Может я стучусь не потому порту?

**Comment by vvip-68** [comment]
Да нет, порт, вроде 80 используется.
Есть ли у вас ограничения в пробросе по протоколу?
Ну, то есть сама страничка рисуется по HTTP, но вот для обмена данными используется WebSocket.
Хотя он вроде тоже на 80-м порту, но может быть есть какие-то нюансы?

**Comment by kex29** [comment]
Такая же проблема! У кого-то получилось пробросить и получить доступ к веб интерфейсу?

**Reply by vvip-68** [reply]
Не будет работать. Мало получить картинку веб-интерфейса на экране смартфона. Управление ведется через соединение по Web-сокету. Который в локальной сети работает, а при соединении с внешними сетями браузер блокирует незащищенный сокет ws и требует шифрованное соединение wss.

Микроконтроллер для этого слишком слабая штучка, чтобы поддерживать сертификаты и шифрование трафика, да и ресурсов у него немного. Да и автор с этим особо не разбирался. В общем через проброс портов работать не будет. 

Если вам реально требуется удаленное управление и дома в локалке где крутится контроллер есть постоянно включенный комп - рекомендуется и на компе и на телефоне установить какой-нибудь Remote Desktop.

Люди писали что с этим отлично справляется плагин от гугла, встроенный в браузер Chrome. Лично у меня с ним не сложилось. Я рекомендую поставить что-то вроде "Ассистент" - https://мойассистент.рф/скачать
Я им пользуюсь. Нормально работает. Не требует белых адресов и проброса порта, может работать за Firewall, поскольку весь трафик  идет через их сервера.

**Reply by GeorgyBenelli** [reply]
У меня через пророс в облако кинетика страницу открывает.Но вечное Подключение...
Причем тот же WLED отлично во внешку пробрасывается на той же есп.Поэтому очень надо MQTT) Хоть так проблему решить;)

**Reply by vvip-68** [reply]
@GeorgyBenelli почему не работает - описано выше. WLED очевидно использует другой путь коммуникации. Раз он работает - вам лучше остановиться на нем.

С MQTT проблема в том, что никто не может сформулировать как именно оно должно работать.
"Включать/Выключать лампу, переключать эффекты и т.д" - это не формулировка и не описание технической стороны вопроса, технических аспектов реализации. Это общие пожелания. За которыми стоит "Дайте инструкцию где что прописать, куда нажать, чтобы оно всё заработало".

Пока никто не может четко сказать что требуется - оно и не будет реализовано.
Вы можете конкретно сформулировать ТЗ, а потом, когда / если случится реализация - все это оттестировать?

**Comment by GeorgyBenelli** [comment]
> @GeorgyBenelli почему не работает - описано выше. WLED очевидно использует другой путь коммуникации. Раз он работает - вам лучше остановиться на нем.
> 
> С MQTT проблема в том, что никто не может сформулировать как именно оно должно работать. "Включать/Выключать лампу, переключать эффекты и т.д" - это не формулировка и не описание технической стороны вопроса, технических аспектов реализации. Это общие пожелания. За которыми стоит "Дайте инструкцию где что прописать, куда нажать, чтобы оно всё заработало".
> 
> Пока никто не может четко сказать что требуется - оно и не будет реализовано. Вы можете конкретно сформулировать ТЗ, а потом, когда / если случится реализация - все это оттестировать?

>вам лучше остановиться на нем
Ну не)Влед за 4 года чето поднадоел.Хотя для тех же панелей там очень крутые эффекты.Ваш проект поинтереснее будет.

>Вы можете конкретно сформулировать ТЗ
Я могу и хочу.Но понимать бы что за ТЗ нужно.Например я могу составить карту топиков и команд.Тестировать смогу конечно.И не я один и не на одной системе умного дома.

**Reply by vvip-68** [reply]
Вот, карту топиков. Что в каком формате и куда посылать, с каких топиков что слушать. Хотя бы с этого можно начать.
Желательно еще и сервер с которым это должно работать. Потому что большинство бесплатных MQTT-брокеров имеют существенные ограничения - например на более 1000 запросов в сутки. Или не более 1 запроса в секунду. 

Захотите включиться - давайте подробнее обсуждать. Ну, например, в Телеграмм. @vvip_68

**Reply by GeorgyBenelli** [reply]
Договорились.Я накидаю такую карту,а Вы, если она вам понравится - реализуете)
Я за все время использовал 5 разных систем умного дома с опенсурс и коммерции, примерно представляю что подходит.

По поводу сервера... В моем понимании все должно работать локально.Я про MQTT говорю с точки зрения умного дома.А если это умный дом с MQTT то он умеет быть и сервером/брокером MQTT.А уж со внешки к этим системам есть подключение.

Поищу бесплатные MQTT сервера, но думаю там все будет в ограничениях

**Reply by vvip-68** [reply]
Ну большинство из тех, кто хочет MQTT, полагаю, не умеют ставить домашние сервера и в них брокеров. И я в том числе. Им просто нужно удаленное управление. И большинство, полагаю - работает под виндой. И я в том числе.

Тогда с вас еще инструкция как и какой поставить и настроить брокер в домашней сети под винду, пробросить управление извне. Линух ради этого я себе ставить не буду. И скорее всего вообще снесу брокер после завершения разработки. Потому что мне это не нужно и пользоваться этим я все равно не буду.

**Reply by GeorgyBenelli** [reply]
Ну брокер по классике это Mosquitto.Как на винду его поставить есть куча видео, но для вики надо будет инструкцию состряпать.Это не проблема
Но конечно напрямую пользоваться этим - тот еще самодазахизм;)Укажи топик в приложении смартфона и отправь команду и тд...

**Reply by vvip-68** [reply]
Ну, для проверки / отладки подойдет. Хорошо, поищу-поставлю мокита. Но скорее уже после нового года.
Пока нет времени особо и есть незавершенные дела... И вам время - сформулировать, описать, сделать инструкцию...

**Comment by GeorgyBenelli** [comment]
А так-по железу минимального сервера для дома это 2000р+-.Тот же тв бокс H96 Max с али.На него отлично дебиан встает

**Reply by vvip-68** [reply]
Ну, инструкцию где купить и как настроить - написать можно. Я точно себе это покупать не буду. Я бы предпочел обойтись имеющимся оборудованием и виндой.

**Reply by GeorgyBenelli** [reply]
ну я то мог бы написать инструкцию.Но это ж будет история про то как организовать свой мктт сервер а не про то как дистанционно к нему подключиться)
Людям без серверов своих подойдет больше облачный мктт брокер думаю.

**Reply by vvip-68** [reply]
Ну вот, что можете - то и описывайте.
Я все равно не смогу грамотно все это описать, потому что не в теме.
Плюс к этому - я и сам воспользуюсь этой инструкцией, чтобы настроить среду для разработки и частичного тестирования.

Не будет инструкции как к нему дистанционно подключиться - значит все будет работать только в локальной сети. И никто кроме вас / вашего сценария использования - этим пользоваться не сможет.
